python/lsst/ts/externalscripts/auxtel/latiss_acquire.py

```diff
@@ -0,0 +1,577 @@
+                )
+
+            self.log.debug(
+                f"At end of iteration loop {iter_num+1}, success is {_success}."
+            )
+
+        # Check that maximum number of iterations for acquisition
+        # was not reached
+        if not _success:
+            self.log.debug(
+                "Failed to acquire star on target after "
+                f"{iter_num} images. Removing focus offset and "
+                "raising an exception"
+            )
+
+            raise RuntimeError(
+                f"Failed to acquire star on target after {iter_num} images."
+            )
+
+    async def latiss_slew_and_acquire(self):
+        # Setup instrument and telescope
+        # The following code sets up the instrument and slews the telescope
+        # simultaneously.
+
+        # If coordinates are provided then use those and set the target name.
+        if self.object_ra and self.object_dec:
+            self.log.debug("Using slew_icrs (object coordinate designation)")
+            _slew_coro = self.atcs.slew_icrs(
+                self.object_ra,
+                self.object_dec,
+                target_name=self.object_name,
+                rot=self.rot_value,
+                rot_type=getattr(RotType, self.rot_type),
+                slew_timeout=240,
+                az_wrap_strategy=self.azimuth_wrap_strategy,
+                time_on_target=self.time_on_target,
+            )
+        # Otherwise use the target name to query Simbad for coordinates.
+        else:
+            self.log.debug("Using slew_object (object name designation)")
+            _slew_coro = self.atcs.slew_object(
+                name=self.object_name,
+                rot=self.rot_value,
+                rot_type=getattr(RotType, self.rot_type),
+                slew_timeout=240,
+                az_wrap_strategy=self.azimuth_wrap_strategy,
+                time_on_target=self.time_on_target,
+            )
+
+        tmp, data = await asyncio.gather(
+            _slew_coro,
+            self.latiss.setup_atspec(grating=self.acq_grating, filter=self.acq_filter),
+        )
+
+        # Perform acquisition loop to center target.
+        await self.execute_acquisition(self.target_position)
+
+        # Verify with another image that we're on target?
+        if self.target_pointing_verification:
+            await self.latiss.take_acq(
+                exptime=self.acq_exposure_time,
+                n=1,
+                group_id=self.group_id,
+                reason=self.reason,
+                program=self.program,
+            )
+        else:
+            self.log.debug(
+                f"Skipping additional image to verify offset was applied correctly as "
+                f"target_pointing_verification is set to {self.target_pointing_verification}"
+            )
+
+    async def latiss_reacquire(self):
+        await self.latiss.setup_atspec(grating=self.acq_grating, filter=self.acq_filter)
```

Comment:
I think the setup was already done during the slew no?

Reply:
Not necessarily, I guess with the "re-acquire" I was imaging somebody would want to use this script in the middle of a block to reposition (or check the position) of their target. In that case, they might want a different latiss configuration than the initial slew.

Reply:
uhm.. so maybe only do this in this condition? e.g. when not slewing.

Reply:
I added a check on the current latiss setup to accommodate this case.